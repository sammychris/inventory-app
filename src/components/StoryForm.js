import styled from 'styled-components';
import FieldLabel from './FieldLabel';
import SelectInputField from './SelectInputField';
import TextAreaField from './TextAreaField';
import Button from './Button';
import CancelButton from './CancelButton';

const StoryForm = ({ bookId, storyField, onClose, onBookIdChange, onStoryChange, onSubmit, books}) => {
    return (
        <StoryFormContainer>
            <CancelButton event={onClose}>close</CancelButton>
            <FieldLabel title="Select a book category" />
            <SelectInputField event={onBookIdChange} bookId={bookId} books={books}/>
            <TextAreaField value={storyField} event={onStoryChange} />
            <div>
                <Button event={onSubmit}>Save</Button>
            </div>
        </StoryFormContainer>
    );
  }


export default StoryForm;

const StoryFormContainer = styled.div`
    background: whitesmoke;
    padding: 20px;
    position: relative;
`;
