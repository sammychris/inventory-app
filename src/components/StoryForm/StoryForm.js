import styled from 'styled-components';
import FieldLabel from '../FieldLabel';
import SelectInputField from '../SelectInputField';
import TextAreaField from '../TextAreaField';
import Button from '../Button';

const StoryForm = ({ bookId, storyField, onClose, onBookIdChange, onStoryChange, onSubmit, books}) => {
    return (
        <StoryFormContainer>
            <Button cls="danger" position="right" size="small" event={() => onClose('story')}>close</Button>
            <FieldLabel title="Select a book category" />
            <SelectInputField event={onBookIdChange} bookId={bookId} books={books}/>
            <TextAreaField value={storyField} event={onStoryChange} />
            <div>
                <Button event={onSubmit}>Save</Button>
            </div>
        </StoryFormContainer>
    );
  }


export default StoryForm;

const StoryFormContainer = styled.div`
    background: whitesmoke;
    padding: 20px;
    position: relative;
`;
