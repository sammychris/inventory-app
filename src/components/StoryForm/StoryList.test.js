import { shallow } from "enzyme";
import StoryForm from "./StoryForm";

test("expect to render a StoryForm", () => {
  const props = { 
    bookId: '0',
    storyField: 'sammy',
    onClose: jest.fn(),
    onBookIdChange: jest.fn(),
    onStoryChange: jest.fn(),
    onSubmit: jest.fn(),
    books: []
  };
  const wrapper = shallow(<StoryForm {...props} />);
  expect(wrapper.length).toEqual(1);
});

test("expect to make a snap shot for StoryForm component", () => {
  const props = {
    bookId: '0',
    storyField: 'sammy',
    onClose: jest.fn(),
    onBookIdChange: jest.fn(),
    onStoryChange: jest.fn(),
    onSubmit: jest.fn(),
    books: []
  };
  const wrapper = shallow(<StoryForm {...props} />);
  expect(wrapper).toMatchSnapshot();
});