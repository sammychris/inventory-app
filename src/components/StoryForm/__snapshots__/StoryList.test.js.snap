// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to make a snap shot for StoryForm component 1`] = `ShallowWrapper {}`;
